src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php

```diff
@@ -16,28 +16,34 @@
     /**
      * Update the model's update timestamp.
      *
+     * @param  mixed  $column
      * @return bool
      */
-    public function touch()
+    public function touch($column = null)
     {
         if (! $this->usesTimestamps()) {
             return false;
         }
 
-        $this->updateTimestamps();
+        $this->updateTimestamps($column);
 
         return $this->save();
     }
 
     /**
      * Update the creation and update timestamps.
      *
+     * @param  mixed  $column
      * @return void
      */
-    protected function updateTimestamps()
+    protected function updateTimestamps($column = null)
     {
         $time = $this->freshTimestamp();
 
+        if ($column) {
```

Comment:
You should probably also check that `$colum` is a date:
`if ($column && $this->isDateAttribute($column)) {`